cmd/template/canary.go

```diff
@@ -214,3 +217,27 @@ func buildWeightStepNode(value string, comment string) *yaml.Node {
 	weight.Content = append(weight.Content, weightNode, weightValuesNode)
 	return weight
 }
+
+func buildExposeServiceStepNode(serviceNames []string, comment string) *yaml.Node {
+	preview := &yaml.Node{Kind: yaml.MappingNode, Tag: "!!map"}
+	previewNode, previewValuesNode := util.BuildMapNode("exposeServices", comment)
+	servicesNode, servicesValuesNode := util.BuildSequenceNodeWithTailComment("services", "")
+	for _, svcName := range serviceNames {
+		node := &yaml.Node{
+			Kind:  yaml.ScalarNode,
+			Tag:   "!!str",
+			Value: svcName,
+		}
+		servicesValuesNode.Content = append(servicesValuesNode.Content, node)
+	}
+
+	ttlNode, ttlNodeValue := util.BuildMapNode("ttl",
+		"Optional. Sets lifetime of the exposed service preview. After that period service preview automatically expires. Max lifetime cannot exceed 24 hours.")
```

Comment:
I thought we had discussed making this required (so it would communicate that its temporary) ... did we decide to keep it optional? 🤔

Reply:
i can't recall discussion like that tbh... anyway - it is optional in deploy engine with a default of 5 minutes...